backend/dataall/cdkproxy/stacks/sagemakerstudio.py

```diff
@@ -146,14 +146,41 @@ def create_sagemaker_domain_resources(self, sagemaker_principals):
             'SagemakerDomainKmsKey',
             alias='SagemakerStudioDomain',
             enable_key_rotation=True,
+            admins=[
+                iam.ArnPrincipal(self.environment.CDKRoleArn)
+            ],
             policy=iam.PolicyDocument(
                 assign_sids=True,
                 statements=[
                     iam.PolicyStatement(
-                        resources=['*'],
+                        actions=[
+                            "kms:Encrypt",
+                            "kms:Decrypt",
+                            "kms:ReEncrypt*",
+                            "kms:GenerateDataKey*",
+                            "kms:CreateGrant"
+                        ],
+                        effect=iam.Effect.ALLOW,
+                        principals=[
+                            sagemaker_domain_role,
+                            iam.ArnPrincipal(self.environment.CDKRoleArn)
```

Comment:
Do we need the cdk role here or just as keyadmin?

Reply:
The `CreateDomain` Operation requires the creator to have the following permissions on the CMK Key:
- kms:CreateGrant
- kms:Decrypt
- kms:DescribeKey
- kms:GenerateDataKeyWithoutPlainText

Similarly `kms:CreateGrant` is needed along with Encrypt/Decrypt to start a number of ML related Jobs in Studio if specifying the KMS Key